### PR TITLE
Deploy&Run value validation

### PR DESCRIPTION
**Pull request by LianaHus**
fix https://github.com/ethereum/remix-project/issues/565

### CODE REVIEW

File: apps/remix-ide/src/app/tabs/runTab/settings.js [hunk 1]
```diff
@@ -53,6 +53,17 @@ class SettingsUI {
     })
   }
 
+  validateValue () {
+    this.el.querySelector('#value').value = parseInt(this.el.querySelector('#value').value)
```
Comment (ioedeveloper) [inline apps/remix-ide/src/app/tabs/runTab/settings.js:57]: Why don't you create a variable for `this.el.querySelector('#value').value` and use the variable instead for your logic.

File: apps/remix-ide/src/app/tabs/runTab/settings.js [hunk 2]
```diff
@@ -53,6 +53,17 @@ class SettingsUI {
...
+    // assign 0 if given value is
+    // - empty
+    // - not valid (for ex 4345-54)
+    // - contains only '0's (for ex 00..0)
+    if (!this.el.querySelector('#value').value) this.el.querySelector('#value').value = 0
+    // if giveen value is negative, ignore '-'
+    if (this.el.querySelector('#value').value < 0) this.el.querySelector('#value').value = Math.abs(this.el.querySelector('#value').value)
```
Comment (ioedeveloper) [inline apps/remix-ide/src/app/tabs/runTab/settings.js:64]: You should probably convert `this.el.querySelector('#value').value` to number before making comparison to zero (0), just to be safe.

Comment (LianaHus) [reply apps/remix-ide/src/app/tabs/runTab/settings.js:64]: it is a number. the types of input is number

File: apps/remix-ide/src/app/tabs/runTab/settings.js (hunk 1 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/tabs/runTab/settings.js:57]: I'm changing the value in each step

**Review by ioedeveloper — APPROVED** [review]

File: apps/remix-ide-e2e/src/tests/runAndDeploy.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 'use strict'
+import { doesNotThrow } from 'assert'
```
Comment (ioedeveloper) [inline apps/remix-ide-e2e/src/tests/runAndDeploy.ts:2]: This line doesn't seem to be needed.

File: apps/remix-ide/src/app/tabs/runTab/settings.js [hunk 4]
```diff
@@ -53,6 +53,28 @@ class SettingsUI {
     })
   }
 
+  validateInputKey (e) {
+    // preventing not numeric keys
+    // preventing 000 case
+    if (!helper.isNumeric(e.key) ||
+      (e.key === '0' && !parseInt(this.el.querySelector('#value').value) && this.el.querySelector('#value').value.length > 0)) {
```
Comment (ioedeveloper) [inline apps/remix-ide/src/app/tabs/runTab/settings.js:60]: Is e.key === '0' consistent for all browsers? You can also have this (`this.el.querySelector('#value')` ) in a variable to make it more readable.

**Review by ioedeveloper — CHANGES_REQUESTED** [review]

File: apps/remix-ide/src/app/tabs/runTab/settings.js (hunk 4 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/tabs/runTab/settings.js:60]: it is

**Review by ioedeveloper — APPROVED** [review]

**Pull request merged**